Modules/BenchPress.Azure/Public/CustomAssertions/BeInResourceGroup.ps1

```diff
@@ -0,0 +1,60 @@
+function ShouldBeInResourceGroup ($ActualValue, [string]$ExpectedValue, [switch] $Negate, [string] $Because) {
+  <#
+    .SYNOPSIS
+      Custom Assertion function to check resource's resource group.
+
+    .DESCRIPTION
+      BeInResourceGroup is a custom assertion that checks the resource group of a resource.
+      It can be used when writing Pester tests.
+
+    .EXAMPLE
+      $result = Confirm-AzBPContainerRegistry -ResourceGroupName $rgName
+
+      $result | Should -BeInResourceGroup 'testrg'
+
+    .EXAMPLE
+      $result = Confirm-AzBPContainerRegistry -ResourceGroupName $rgName
+
+      $result | Should -Not -BeInResourceGroup 'testrg2'
+
+    .INPUTS
+      ConfirmResult
+      System.Switch
+      System.String
+
+    .OUTPUTS
+      PSCustomObject
+  #>
+  if ($null -eq $ActualValue){
+    [bool] $succeeded = $false
+
+    if ($Negate) { $succeeded = -not $succeeded }
+    $failureMessage = "Confirm result is null or empty."
+  }
+  else {
+    $resourceGroupName = $ActualValue.ResourceDetails.ResourceGroupName
```

Comment:
Same here, just confirm that ResourceGroupName is always at the root of the returned resource object regardless of the type.

Reply:
Will deal with this in next issue #207